packages/ui-components/src/components/Inputs/HexcodeField.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { FormHelperText } from '@material-ui/core';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { TextField } from './TextField';
+
+const HexcodeFieldWrapper = styled.div`
+  margin-bottom: 1em;
+`;
+
+const HexcodeFieldContainer = styled.div`
+  display: flex;
+  align-items: flex-end;
+  margin-top: 1em;
+`;
+
+const HexcodePreview = styled.div`
+  background-color: ${({ backgroundColor }) => backgroundColor};
+  height: 100%;
+  width: 100%;
+  border-radius: 50%;
+  width: 3em;
+  height: 3em;
+  margin-left: 1em;
+  margin-bottom: 0.2em;
+  border: ${({ theme }) => `1px solid ${theme.palette.common.black}`};
+`;
+
+const HexInput = styled(TextField)`
+  max-width: 18em;
```

Comment:
```suggestion
  max-width: 18em;
```
This seems like a sensible value but we probably don't want to hard code this right? Better to leave it to the parent to determine.